hello.py

```diff
@@ -1,33 +1,69 @@
-from flask import Flask, render_template
+from flask import Flask, render_template, jsonify
 from flask_nav import Nav
-from flask_nav.elements import Navbar, View
+from flask_nav.elements import Navbar, View, Subgroup
+import json
 
 app = Flask(__name__)
 nav = Nav(app)
 
-nav.register_element("top", Navbar(
+with open("database.json", "r") as d:
+  data = json.load(d)
+
+categories = sorted(set([item["category"] for item in data]))
+categories_sub = Subgroup("Categories")
+
+for cat in categories:
+    categories_sub.items.append(View(cat, "category", cat = cat))
+
+topbar = Navbar(
     "Top_bar",
     View("Home", "index"),
-    View("Categories", "category"),
+    View("All Products", "all_cats"),
+    categories_sub,
     View("About Mobius", "about")
-))
+)
+
+nav.register_element("top", topbar)
 
 @app.route("/")
 def index():
     return render_template("index.html")
 
-@app.route("/product")
-def product():
-    return render_template("product.html")
+@app.route("/product/<name>")
+def product(name):
+    item_data = get_item_from_data(name, data)
```

Comment:
_(Optional)_ You could use the [dictionary unpacking operator](https://python-reference.readthedocs.io/en/latest/docs/operators/dict_unpack.html)`**`, as long as the names used in the template (html file) matched with what's in `database.json`. So you could do `return render_template("product.html", **item_data)`.